power_grid_model_io_native_c/power_grid_model_io_native/include/power_grid_model_io_native/pgm_vnf_converter/converter_parser.hpp

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_IO_NATIVE_C_CONVERTER_PARSER_HPP
+#define POWER_GRID_MODEL_IO_NATIVE_C_CONVERTER_PARSER_HPP
+
+#include <power_grid_model_io_native/common/common.hpp>
+
+#include <power_grid_model/auxiliary/input.hpp>
+#include <power_grid_model/container.hpp>
+
+#include <iostream>
+#include <map>
+#include <regex>
+
+namespace power_grid_model_io_native {
+using VisionGUID = std::string;
+
+struct VnfNode {
+    VisionGUID guid{"nan"};
+    double u_nom{nan};
+};
+
+using VnfGrid = pgm::Container<VnfNode>;
+template <typename Identifier> using IdentifierLookup = std::map<Identifier, ID>;
+using VisionGUIDLookup = IdentifierLookup<VisionGUID>;
+
+class PgmVnfParser {
+  public:
+    explicit PgmVnfParser(std::string_view vnf_data);
+
+    void parse_input();
+    VisionGUIDLookup get_id_lookup() const;
+    VnfGrid get_parsed_data() const;
+
+  private:
+    VisionGUIDLookup vision_guid_lookup_;
+    std::string_view vnf_data_;
+    VnfGrid vnf_parsed_data_;
+    ID id_count_{0};
+
+    std::string_view find_node_block() const;
+    void parse_node_input();
+};
+
+inline PgmVnfParser::PgmVnfParser(std::string_view const vnf_data) : vnf_data_(vnf_data) {}
+
+inline void PgmVnfParser::parse_input() {
+    // parse each component individually and finish constructing the container
+    parse_node_input();
+    this->vnf_parsed_data_.set_construction_complete();
+}
+
+inline std::string_view PgmVnfParser::find_node_block() const {
+    // obtain the whole string block for nodes
+    std::regex const nodes_regex{R"(\[NODE\]([\s\S]*?)\[\])"};
+    std::match_results<std::string_view::const_iterator> nodes_match;
+
+    std::regex_search(this->vnf_data_.begin(), this->vnf_data_.end(), nodes_match, nodes_regex);
+    auto const& nodes = nodes_match[1]; // the first group is the nodes data
```

Comment:
fallback that returns an empty `string_view` (`""sv`) or `assert` that the size `> 1`